src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.h

```diff
@@ -405,6 +409,33 @@ class ClusterQueueHelper BSLS_KEYWORD_FINAL
     /// queue which have a proper valid unique queueId.
     typedef bsl::unordered_map<int, QueueContext*> QueueContextByIdMap;
 
+    class QueueContextIterator {
+      private:
+        QueueContextMap& d_base;
+
+        QueueContextMap::const_iterator d_current;
+
+        bmqp_ctrlmsg::Status d_status;
+
+      public:
+        QueueContextIterator(QueueContextMap&  base,
+                             bslma::Allocator* allocator_p);
+
+        QueueContextIterator(QueueContextMap&                       base,
+                             const QueueContextMap::const_iterator& cit,
+                             bslma::Allocator* allocator_p);
+
+        ~QueueContextIterator();
+
+        void next();
+
+        bmqp_ctrlmsg::Status* status();
+
+        bool atEnd() const;
+
+        const mqbblp::ClusterQueueHelper::QueueContextSp& queueContext();
```

Comment:
Might place iteration-related functions together and getters together.
```suggestion
        bmqp_ctrlmsg::Status* status();
        const mqbblp::ClusterQueueHelper::QueueContextSp& queueContext();
        
        void next();
        bool atEnd() const;
```